src/components/Copy/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { faCopy } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Box, Tooltip, useTheme } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+
+export interface CopyProps {
+	copy: string | undefined;
+	onClick?: () => void;
+}
+
+export interface CopyPropsWithChildren extends CopyProps {
+	children: React.ReactNode;
+	variant?: 'absolute' | 'adjacent';
+}
+
+export const copyToClipboard = async (text: string): Promise<void> => {
+	try {
+		await navigator.clipboard.writeText(text);
+	} catch (err) {
+		console.error('Failed to copy text: ', err);
+	}
+};
+
+export const Copy = ({
+	copy,
+	onClick,
+	...props
+}: CopyProps | CopyPropsWithChildren) => {
+	let closeTooltipTimeout: ReturnType<typeof setInterval> | null = null;
```

Comment:
this cannot be a variable it must be a state, if the component re-render you will loose the information

Reply:
eg:


```
const [tooltipVisible, setTooltipVisible] = useState(false);
  const [closeTooltipTimeout, setCloseTooltipTimeout] = useState(null);

  // Function to show the tooltip
  const showTooltip = () => {
    setTooltipVisible(true);

    // Set timeout to automatically hide tooltip after 5 seconds
    const timeoutId = setTimeout(() => {
      setTooltipVisible(false);
      setCloseTooltipTimeout(null); // Reset timeout state after firing
    }, 5000);

    setCloseTooltipTimeout(timeoutId);
  };

  // Effect to clear timeout if closeTooltipTimeout changes or component unmounts
  useEffect(() => {
    return () => {
      if (closeTooltipTimeout) {
        clearTimeout(closeTooltipTimeout);
      }
    };
  }, [closeTooltipTimeout]);

```